### PR TITLE
Minor fixes to the EVP model

### PR DESCRIPTION
**Pull request by feliperodri**
Signed-off-by: Felipe R. Monteiro <felisous@amazon.com>

*Issue #, if available:*

N/A.

*Description of changes:*

Minor fixes to the `EVP` model.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/evp_override.c [hunk 1]
```diff
@@ -1029,3 +1027,8 @@ void evp_md_ctx_shallow_free(EVP_MD_CTX *ctx) {
     free(ctx);
     // Does not free EVP_KEY, since this is always done separately in our use cases
 }
+
+void EVP_MD_CTX_set_flags(EVP_MD_CTX *ctx, int flags) {
+    assert(__CPROVER_r_ok(ctx, sizeof(*ctx)));
```
Comment (SaswatPadhi) [inline source/evp_override.c:1032]: Should this be `__CPROVER_w_ok` instead? We're modifying the `flags` field below ...

**Review by SaswatPadhi — COMMENTED** [review]
LGTM. Just one question below:

File: source/evp_override.c (hunk 1 above)
Comment (feliperodri) [reply source/evp_override.c:1032]: Good catch! Thanks! Implemented as suggested.

**Review by SaswatPadhi — APPROVED** [review]
Thanks for the quick fix! LGTM.

**Pull request merged**